packages/ui/core-components/src/lib/atoms/inputs/dropdown/Dropdown.svelte

```diff
@@ -39,162 +38,199 @@
+			// We need a canonical way / function that handles query replacement like this (e.g. updating buildReactiveInputQuery)
+			const searchQ = query.search(search, 'label');
+			await searchQ.fetch();
+			queryOptions = searchQ;
 		} else {
-			ctx.hasBeenSet = false;
+			queryOptions = query;
 		}
-	}
+	}, 250);
 
-	let items;
-	$: useNewQuery($query);
+	// Update the items when the query changes
+	$: search, query, updateQueryOptions();
 
-	const debouncedUpdateItems = debounce(updateItems, 250);
-	$: debouncedUpdateItems(search);
-</script>
+	/**
+	 * Resets the defaults whenever parameters change
+	 */
+	function evalDefaults() {
+		resolveMaybePromise(
+			() => {
+				if ($selectedOptions.length) {
+					// Try to keep any current selection before falling back to defaults
+					// e.g. if the passed-in query has changed, but the user had a previous selection
+					const presentValues = $selectedOptions.filter((x) =>
+						$options.some((o) => o.value === x.value && o.label === x.label)
+					);
+					// If the available values have changed (e.g. one is missing in the new query), remove that one and update the selection
+					if (presentValues.length !== $selectedOptions.length) $selectedOptions = presentValues;
+					// If there is at least one selection remaining, then we don't need to revert to defaults
+					if ($selectedOptions.length) return;
+				}
+				if (noDefault)
+					return; // noop
+				else if (typeof defaultValue !== 'undefined' && defaultValue !== null) {
+					// hard-coded default
+					const _def = Array.isArray(defaultValue) ? defaultValue : [defaultValue];
+					if (_def.length) {
+						const presentDefaults = $options.filter((x) => _def.some((d) => x.value === d));
+						// const missingDefaults = _def.filter((d) => !presentDefaults.some((x) => x.value === d));
+						if (presentDefaults.length) {
+							$selectedOptions = [...presentDefaults];
+						}
+						return;
+					}
+				}
+				// Default Behavior: set to the first value
+				if (!multiple && $selectedOptions.length) $selectedOptions = [$options[0]];
+			},
+			// Fetch the query, then wait for DOM updates to flush (this ensures that the options store is up to date)
+			() => (query ? query.fetch().then(() => tick()) : tick()),
+			(err) => {
+				console.error(`Error while updating Dropdown Query: ${err.message}`);
+			}
+		);
+	}
 
-{#key $items}
-	<Invisible>
-		<slot />
+	$: $query, $options, evalDefaults();
+</script>
 
-		{#if hasQuery && $items.length > 0 && $items.loaded}
-			<DropdownOption value={$items[0].value} valueLabel={$items[0].label} />
-		{/if}
-	</Invisible>
+<slot />
+<!-- 
+	Key is needed here to ensure that we re-render the options list 
+	If we don't do this, sorting is not properly updated
+-->
+{#key queryOptions}
+	<QueryLoad data={$queryOptions} let:loaded>
+		<!-- TODO: Add temporary option that shows loading state -->
+		{#each loaded?.sort((a, b) => a.idx - b.idx) ?? [] as queryOpt, i (queryOpt.value)}
```

Comment:
maybe switch to 
```js
(`${queryOpt.label} | ${queryOpt.value}`)
```
I don't think values are necessarily guaranteed to be unique